dotcom-rendering/src/components/FormatBoundary.tsx

```diff
@@ -0,0 +1,37 @@
+import { css } from '@emotion/react';
+import type { ArticleFormat } from '@guardian/libs';
+import { paletteDeclarations } from '../palette';
+import { useConfig } from './ConfigContext';
+
+type Props = {
+	format: ArticleFormat;
+	children: React.ReactNode;
+};
+
+/**
+ * Used to create a set of palette colours for a different format
+ * to that of the :root element
+ */
+export const FormatBoundary = ({ format, children }: Props) => {
+	const { darkModeAvailable } = useConfig();
+
+	return (
+		<div
+			data-format-theme={format.theme}
+			data-format-design={format.design}
+			data-format-display={format.display}
+			css={[
+				css`
+					${css(paletteDeclarations(format, 'light'))}
+					@media (prefers-color-scheme: dark) {
+						${darkModeAvailable
+							? css(paletteDeclarations(format, 'dark'))
+							: '/* dark mode unavailable */'}
+					}
```

Comment:
We probably want to wrap this in a `renderingTarget` check to prevent dark mode appearing on dotcom? It might be worth doing the whole check outside the `prefers-color-scheme` media query so that nothing appears in the CSS if either a) on dotcom or b) on apps with dark mode turned off?

This is where we do this for the root:

https://github.com/guardian/dotcom-rendering/blob/6ee9f6375b091becf7a99d5e3f129ac92d5540a9/dotcom-rendering/src/components/ArticlePage.tsx#L67-L73

Reply:
The value of `darkModeAvailable` is coming from the `Config`. Whether it’s bound to a rendering target or a switch is an implementation detail that I thought would be best abstracted upstream.

I agree however that we should not ship entirely unused CSS, I thought Emotion would have been able to tree-shake this declaration out, but testing it out it does not seem like it:

```html
```

Reply:
> The value of `darkModeAvailable` is coming from the `Config`. Whether it’s bound to a rendering target or a switch is an implementation detail that I thought would be best abstracted upstream.

Fair enough, but in that case, given `darkModeAvailable` is added in this PR, would it be best to include a check for `renderingTarget` in `darkModeAvailable` now? At the moment it only depends on the value of the switch, and therefore I don't think there's anything that will prevent these styles being applied on dotcom once that switch is flipped to `true`?

Reply:
Maybe it’s [easier to see with a strict boolean](https://github.com/guardian/dotcom-rendering/compare/mxdvl/format-boundary...darkmode/fromat-boundary-strict-boolean), but this value would need to be set explicitely in every config. Turning the switch on [will only enable dark mode in apps](https://github.com/guardian/dotcom-rendering/compare/mxdvl/format-boundary...darkmode/fromat-boundary-strict-boolean#diff-c0c5d82647133e5308b928fa6e77fd202abf6e562e12e7bbbbb7f400046509acL23).